rosidl_typesupport_introspection_c/include/rosidl_typesupport_introspection_c/message_introspection.h

```diff
@@ -24,19 +24,41 @@
 
 #include "rosidl_typesupport_introspection_c/visibility_control.h"
 
+/// Structure used to describe a single field of an interface type.
 typedef struct rosidl_typesupport_introspection_c__MessageMember_s
 {
+  /// The name of the field.
   const char * name_;
+  /// The type of the field as a value of the field types enum
+  /// rosidl_typesupport_introspection_c_field_types.
+  /// e.g. rosidl_typesupport_introspection_c__ROS_TYPE_FLOAT
   uint8_t type_id_;
+  /// If the field is a string, the upper bound on the length of the string.
   size_t string_upper_bound_;
+  /// If the type_id_ value is rosidl_typesupport_introspection_c__ROS_TYPE_MESSAGE,
+  /// this points to an array describing the fields of the sub-interface.
   const rosidl_message_type_support_t * members_;
+  /// True if this field is an array type, false if it is any other type. An
+  /// array has the same value for / type_id_.
   bool is_array_;
+  /// If is_array_ is true, this contains the number of members in the array.
   size_t array_size_;
+  /// If is_array_ is true, this specifies if the array has a maximum size. If it is true, the
+  /// value in array_size_ is the maximum size.
   bool is_upper_bound_;
+  /// The bytes into the interface's in-memory representation that this field can be found at.
   uint32_t offset_;
+  /// If the interface has a default value, this points to it.
   const void * default_value_;
+  /// If is_array_ is true, a pointer to a function that gives the size of one member of the array.
   size_t (* size_function)(const void *);
+  /// If is_array_ is true, a pointer to a function that gives a const pointer to the member of the
+  /// array indicated by index.
+  /// Only used for nested interface types.
```

Comment:
Same comment about being used for more than nested interface types

Reply:
Ouch, somehow I though it did not apply in these functions fixed c740f29